crates/snapbox/src/data.rs

```diff
@@ -10,6 +10,8 @@ pub struct Data {
 enum DataInner {
     Binary(Vec<u8>),
     Text(String),
+    #[cfg(feature = "json")]
+    Json(Vec<serde_json::Value>),
```

Comment:
What if we deserialize the data into a common type and do comparisons on that?  We could call the variant `Structured`.

For yaml, toml, and json, I think toml is the superset.

Benefits:
- easy to add new data formats as long as they are subsets
Downsides
- Might be weird that the user's data is in one format and we show it in a different one.  We could mitigate this by tracking the format for display purposes
- Should we allow cross-format comparisons though?

Reply:
I think the downsides outweigh the benefits here

Downsides:
- Restricts data formats to those that can be turned into the superset
- Implementations for `data format -> common` and `common -> data format`
- Cross comparisons (they only make sense if you are explicitly trying to do so)
- Writing changes could be tricky
- A bug for one could be a bug for all
- Wildcards might be tricky (gut feeling but they could be fine)

Benefits:
- One comparison function (easier to verify correctness)
- API for text operations might be cleaner
- Easier implementation of new data types

I could be overstating the downsides and understating the benefits but I think it is better to do separate formats. Implementing a few data formats might show us that a common type is a better idea. This might need more investigation I'm not entirely sure